documents/attribute.go

```diff
@@ -1,41 +1,76 @@
 package documents
 
 import (
-	"math/big"
-	"reflect"
+	"time"
 
 	"github.com/centrifuge/go-centrifuge/crypto"
-	"github.com/centrifuge/go-centrifuge/errors"
+	"github.com/centrifuge/go-centrifuge/utils"
 	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/golang/protobuf/ptypes/timestamp"
 )
 
 // attributeType represents the custom attribute types allowed in models
 type attributeType string
 
-// String string repr
+// String returns the readable name of the attribute type.
 func (a attributeType) String() string {
 	return string(a)
 }
 
+const (
+	// AttrInt256 is the standard integer custom attribute type
+	AttrInt256 attributeType = "int256"
+
+	// AttrDecimal is the standard big decimal custom attribute type
+	AttrDecimal attributeType = "bigdecimal"
+
+	// AttrString is the standard string custom attribute type
+	AttrString attributeType = "string"
+
+	// AttrBytes is the standard bytes custom attribute type
+	AttrBytes attributeType = "bytes"
+
+	// AttrTimestamp is the standard time stamp custom attribute type
+	AttrTimestamp attributeType = "timestamp"
+)
+
+// allowedAttrTypes holds a map of allowed attribute types and their reflect.Type
+var allowedAttrTypes = map[attributeType]struct{}{
```

Comment:
Maybe convert this to a function that returns the map? So it is not modifiable, knowing that documents is a fairly big package.

Reply:
Yeah made this because we seem to create the map everytime we call the function. it is low hanging performance improvement IMO